### PR TITLE
Port controllers

### LINKED ISSUE #1026: Port old controllers to the new controller
/kind feature

**Describe the solution you'd like**
Now that we're moving to kubebuilder v2 we have the new-style controllers, we ned to port the old reconcile over to the new reconcile.

**Anything else you would like to add:**

**Environment:**

/milestone v0.4
/priority critical-urgent

Should probably do this one at a time as this is a major overhaul

### PR DESCRIPTION
**Pull request by chuckha**
**What this PR does / why we need it**:
This PR moves the reconciles from the old location to the kubebuilderv2 location.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1026 

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by chuckha** [comment]
/cc @rudoi 

because we were talking about this on friday

**Comment by vincepri** [comment]
Changes LGTM, only had 2 minor comments.

@chuckha FYI, you have 2 commits with the same message

**Comment by chuckha** [comment]
@vincepri nah, but they are very similar Machine vs Cluster

File: controllers/awscluster_controller.go [hunk 1]
```diff
@@ -18,33 +18,169 @@ package controllers
...
 	"context"
+	"fmt"
+	"time"
 
 	"github.com/go-logr/logr"
-	infrastructurev1alpha2 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha2"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/record"
+	infrav1alpha2 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha2"
```
[the remark this answers is not in the document]
Comment (chuckha) [reply controllers/awscluster_controller.go]: yeah totally possible 👍

**Comment by vincepri** [comment]
Indeed!

**Comment by chuckha** [comment]
@vincepri how does it look now?

**Pull request merged**